lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameClient.java

```diff
@@ -72,6 +73,7 @@ public NameClient(final String serverAddr,
    * @param factory    an identifier factory
    * @param cache      a cache
    */
+  @Deprecated
```

Comment:
Similar comments for other NameClient's.
Please double check that they're not needed or make sure Tang injects NameClient.